### PR TITLE
Add Homepage News Feed

### PR DESCRIPTION
**Pull request by chrismekim**
### Description
- Adds Homepage News Feed
- UPDATE: This is now a Markdown Text field in Accentuate. 

### Type(s) of changes

- [x] New feature

### Motivation for PR

#9 
### How Has This Been Tested?

https://2qdzpaikvh3moqj3-52674822324.shopifypreview.com
pw: nunu

### Applicable screenshots:

image
image

### Follow-up PR

Pt 2: Add the News page. #31 
- Add News Feed Limit field, so they can choose how many rows to show. It is currently hardcoded to be 4 on the homepage.

### CODE REVIEW

**Comment by sepowitz** [comment]
Not seeing this on the preview link? Seems like the theme's keep getting overridden. Are you killing you watch command before switching branches?

**Review by sepowitz — COMMENTED** [review]
Looks gorg! One thing to note on mobile:

With the overflow-x, we'll probably want to remove the margin on the right of the `news-feed` so that it doesn't cut off and overflow content.

**Comment by chrismekim** [comment]
> Not seeing this on the preview link? Seems like the theme's keep getting overridden. Are you killing you watch command before switching branches?

I think every time I would switch between branches, my .env with the theme ID was changing (?), so whenever I would `yarn deploy` I was constantly deploying different branches!! 💀   

Good catch about the margin right, ty!

**Review by sepowitz — COMMENTED** [review]
I'm still seeing the same issue with the margin on mobile. Also, might be worth asking XXIX, but I assume each row should overflow scroll individually, instead of the container as a whole

**Comment by chrismekim** [comment]
> I'm still seeing the same issue with the margin on mobile. Also, might be worth asking XXIX, but I assume each row should overflow scroll individually, instead of the container as a whole

Ooops, fixed the margin on mobile!
Asked XXIX about the scroll, so waiting for their response :)

**Comment by sepowitz** [comment]
> > Not seeing this on the preview link? Seems like the theme's keep getting overridden. Are you killing you watch command before switching branches?
> 
> I think every time I would switch between branches, my .env with the theme ID was changing (?), so whenever I would `yarn deploy` I was constantly deploying different branches!! 💀
> 
> Good catch about the margin right, ty!

Yup, you need to make sure to update your .env to reflect the theme (branch) you are on. That's why you should always kill your watch command before switching branches, and update you .env before running yarn deploy/watch

**Review by limsohee1002 — APPROVED** [review]

**Comment by chrismekim** [comment]
@muzzbanana FYI this is currently on hold while I wait for XXIX to let me know about the overflow-x scroll :)

File: theme/styles/snippets/news-feed.scss [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        width: 70%;
+        
+        @include media('lg-up') {
+          min-width: 50rem;
+          padding: 0 1.5rem;
+        }
+      }
+  
+      > :nth-child(3)  {
+        width: 20%;
```
Comment (chrismekim) [inline theme/styles/snippets/news-feed.scss:69]: This works but I'm not sure if `first-child`, `nth-child(2)`, `nth-child(3)` is a good way to select these! (I'm selecting the first child (1st col), second child (2nd col), third child (3rd col) of the table)

**Comment by chrismekim** [comment]
@limsohee1002 @sepowitz @muzzbanana The last time you reviewed this was an array of `News Feed Items`, but now the entire News Feed is Markdown Text (XXIX request), and we are just rendering the markdown. I'm applying the same styling as before.

**Question!** I think we may have to have two models, one for `News Feed Homepage`, and one for `News Feed News Page` because `News Feed Homepage` seems to show 4 rows, while the `News Feed News Page` shows a long list of rows. Since they can easily copy and paste the markdown text it seems like that wouldn't be a problem to have two models.. What do you think? 

News Feed on Homepage:
image

News Feed on News Page:
image


This is what the custom field looks like:
image

**Comment by sepowitz** [comment]
> @limsohee1002 @sepowitz @muzzbanana The last time you reviewed this was an array of `News Feed Items`, but now the entire News Feed is Markdown Text (XXIX request), and we are just rendering the markdown. I'm applying the same styling as before.
> 
> **Question!** I think we may have to have two models, one for `News Feed Homepage`, and one for `News Feed News Page` because `News Feed Homepage` seems to show 4 rows, while the `News Feed News Page` shows a long list of rows. Since they can easily copy and paste the markdown text it seems like that wouldn't be a problem to have two models.. What do you think?
> 
> News Feed on Homepage:
> image
> 
> News Feed on News Page:
> image
> 
> This is what the custom field looks like:
> image

This is a cool idea, but I worry this is too loose for editors. Correct me if I'm wrong, but XXIX was asking for markdown so that they could choose whether to render a date or link in the far right column of the news feed item?

I wonder if a better way to model it would be to have:

New Feed model > has many New Feed Items

News Feed Item has:
- Image field
- Body field (This can be markdown)
- Link/Date field (Optional, this can either be part of the body field, or be it's own markdown field; up to you)

* if these are separate fields, we can nix the nth(etc) and just target the specific DOM elements that render these fields (e.g. NewsFeed__img, NewsFeed__body, NewsFeed__link)

As far as having two separate models, I would suggest a single news feed model, with an optional limit field (this can be a number e.g. 4, which we can use on the frontend to limit the number of entries shown, show a read more button or whatever)

**Comment by chrismekim** [comment]
> * Link/Date field (Optional, this can either be part of the body field, or be it's own markdown field; up to you)

@sepowitz Ahhh you're right, the only reason they asked for markdown is the flexibility for the last column to be a link or a date. If I just make that one markdown that should solve the problem!! I don't think they will mind if the entire table is not markdown, but I'll let them know why. :)

File: theme/snippets/news-feed.liquid [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<div class="NewsFeed flex flex-col">
```
Comment (sepowitz) [inline theme/snippets/news-feed.liquid:1]: LG! Only thought is to have the limit come from the news_feed metafield model.

**Review by sepowitz — APPROVED** [review]

**Pull request merged**